dev/system-test/firestore.ts

```diff
@@ -2284,6 +2285,1554 @@ describe('Aggregates', () => {
+      expect(snapshot.data().averagePages).to.equal(100);
+      expect(snapshot.data().countOfDocs).to.equal(2);
+    });
+
+    it('performs aggregations on nested map values', async () => {
+      const testDocs = {
+        a: {
+          author: 'authorA',
+          title: 'titleA',
+          metadata: {pages: 100, rating: {critic: 2, user: 5}},
+        },
+        b: {
+          author: 'authorB',
+          title: 'titleB',
+          metadata: {pages: 50, rating: {critic: 4, user: 4}},
+        },
+      };
+      await addTestDocs(testDocs);
+      const snapshot = await col
+        .aggregate({
+          totalPages: AggregateField.sum('metadata.pages'),
+          averagePages: AggregateField.average('metadata.pages'),
+          averageCriticRating: AggregateField.average('metadata.rating.critic'),
+          totalUserRating: AggregateField.sum('metadata.rating.user'),
+          count: AggregateField.count(),
+        })
+        .get();
+      expect(snapshot.data().totalPages).to.equal(150);
+      expect(snapshot.data().averagePages).to.equal(75);
+      expect(snapshot.data().averageCriticRating).to.equal(3);
+      expect(snapshot.data().totalUserRating).to.equal(9);
+      expect(snapshot.data().count).to.equal(2);
+    });
+
+    it('performs sum that results in float', async () => {
+      const testDocs = {
+        a: {
+          author: 'authorA',
+          title: 'titleA',
+          pages: 100,
+          year: 1980,
+          rating: 5,
+        },
+        b: {
+          author: 'authorB',
+          title: 'titleB',
+          pages: 50,
+          year: 2020,
+          rating: 4.5,
+        },
+        c: {
+          author: 'authorB',
+          title: 'titleB',
+          pages: 150,
+          year: 2021,
+          rating: 3,
+        },
+      };
+      await addTestDocs(testDocs);
+      const snapshot = await col
+        .aggregate({
+          totalRating: AggregateField.sum('rating'),
+        })
+        .get();
+      expect(snapshot.data().totalRating).to.equal(12.5);
+    });
+
+    it('performs sum of ints and floats that results in an int', async () => {
+      const testDocs = {
+        a: {
+          author: 'authorA',
+          title: 'titleA',
+          pages: 100,
+          year: 1980,
+          rating: 5,
+        },
+        b: {
+          author: 'authorB',
+          title: 'titleB',
+          pages: 50,
+          year: 2020,
+          rating: 4.5,
+        },
+        c: {
+          author: 'authorB',
+          title: 'titleB',
+          pages: 150,
+          year: 2021,
+          rating: 3.5,
+        },
+      };
+      await addTestDocs(testDocs);
+      const snapshot = await col
+        .aggregate({
+          totalRating: AggregateField.sum('rating'),
+        })
+        .get();
+      expect(snapshot.data().totalRating).to.equal(13);
+    });
+
+    it('performs sum that overflows max int', async () => {
```

Comment:
I have to fix this test in Web. There will be another PR that will have to be ported for node. I can do this.